code/modules/cm_marines/dropship_equipment.dm

```diff
@@ -860,7 +860,11 @@
 		return
 
 	if(!is_ground_level(selected_stretcher.z)) //in case the stretcher was on a groundside dropship that flew away during our input()
-		return
+		if(!istype(selected_stretcher, /obj/structure/bed/medevac_stretcher/vehicle))
+			return
+		var/obj/structure/bed/medevac_stretcher/vehicle/v_stretcher = selected_stretcher
+		if(!v_stretcher.vehicle || v_stretcher.vehicle.health < (initial(v_stretcher.vehicle.health) * 0.5) || !is_ground_level(v_stretcher.vehicle.z) || v_stretcher.vehicle.next_move > world.time)
+			return
```

Comment:
too many istype checks, write a proc that is run on the stretcher and use inheritance to change behaviour

Reply:
Made a separate proc in stretcher.